scripts/imgtool/image.py

```diff
@@ -328,8 +331,20 @@ def create(self, key, public_key_format, enckey, dependencies=None,
             # Add the size of the TLV info header
             protected_tlv_size += TLV_INFO_SIZE
 
-        # At this point the image is already on the payload, this adds
-        # the header to the payload as well
+        if custom_tlvs:
```

Comment:
This block of code must happen before the previous check for `if protected_tlv_size != 0` above to correctly set the size.

Reply:
Done